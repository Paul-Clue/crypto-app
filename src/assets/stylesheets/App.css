.App {
  width: 100%;
}

.UL {
  position: absolute;
  left: -2.08vw;
  width: 102%;
  list-style-type: none;
}

li {
  list-style-type: none;
}

.navbar {
  border: 5px solid black;
}

.NavHead {
  color: whitesmoke;
}
