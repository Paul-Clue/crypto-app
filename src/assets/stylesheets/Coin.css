.item {
  display: flex;
  justify-content: space-between;
  list-style-type: none;
  border: 5px solid #0ff;
  box-shadow:
    inset 0 0 18px #fff,
    inset -6px 0 18px black,
    inset 6px 0 18px black,
    inset -6px 0 30px black,
    inset 6px 0 30px #0ff,
    0 0 18px #fff,
    4px 0 18px
    #f3bad6,
    -4px 0 18px #0ff;
  width: 100%;
  text-decoration: none;
  color: black;
  cursor: pointer;
}
