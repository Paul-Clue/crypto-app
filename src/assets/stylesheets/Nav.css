.dollar {
  color: gold;
}

.filtTag {
  color: whitesmoke;
  font-weight: bolder;
}

.filtTag2 {
  font-size: small;
}
