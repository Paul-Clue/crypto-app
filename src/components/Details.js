import React from 'react';
// import PropTypes from 'prop-types';
// import { useSelector, useDispatch } from 'react-redux';
// import { useSelector, useDispatch } from 'react-redux';
import { useSelector } from 'react-redux';
import { detailFilter } from '../actions';
import '../assets/stylesheets/Detail.css';

function Details() {
  // const { name } = props;
  // const coin = useSelector((state) => state.addCoin);
  const coinDetail = useSelector((state) => state.detail);
  // const coinDetail2 = useSelector((state) => state.detailFilter);
  // console.log(`This is coinDetail: ${coinDetail.name}`);//eslint-disable-line
  // console.log(`This is detailFilter: ${detailFilter}`);//eslint-disable-line
  // console.log(`This is coin: ${coin}`);//eslint-disable-line
  // const filteredDetails = coin.filter((coin) => coin.name === coinDetail);
  // const dispatch = useDispatch();

  // function showDetail() {
  //   dispatch(detailFilter(filteredDetails));
  // }

  return (
    <>
      <div className="detail">
        <div className="TitleName">
          <img src={coinDetail.icon} alt="coin logo" />
          <h1 className="TitleName1">{coinDetail.name.toUpperCase()}</h1>
        </div>
        <h1 className="detail2">
          <span className="bullet">
            <span className="bold" id="one">Symbol:</span>
            &nbsp;&nbsp;&nbsp;
            {coinDetail.symbol }
          </span>
          <span className="bullet">
            <span className="bold">Rank:</span>
            &nbsp;&nbsp;&nbsp;
            {coinDetail.rank }
          </span>
          <span className="bullet">
            <span className="bold">Price:</span>
            &nbsp;&nbsp;&nbsp;
            {coinDetail.price }
          </span>
          <span className="bullet">
            <span className="bold">Market Cap:</span>
            &nbsp;&nbsp;&nbsp;
            {coinDetail.marketCap }
          </span>
          <span className="bullet">
            <span className="bold">Total Supply:</span>
            &nbsp;&nbsp;&nbsp;
            {coinDetail.totalSupply }
          </span>
          <span className="bullet">
            <span className="bold">Price Change 1h:</span>
            &nbsp;&nbsp;&nbsp;
            {coinDetail.priceChange1h }
          </span>
          <span className="bullet">
            <span className="bold">Price Change 1d:</span>
            &nbsp;&nbsp;&nbsp;
            {coinDetail.priceChange1d }
          </span>
          <span className="bullet">
            <span className="bold">Price Change 1w:</span>
            &nbsp;&nbsp;&nbsp;
            {coinDetail.priceChange1w }
          </span>
        </h1>
      </div>
    </>
  );
}

// Details.propTypes = { name: PropTypes.string.isRequired };

export default Details;
