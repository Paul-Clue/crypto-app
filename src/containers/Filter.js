import React, { useRef } from 'react';
import PropTypes from 'prop-types';
import '../assets/stylesheets/App.css';

function Filter(props) {
  const category = useRef();
  const { filt } = props;
  const { filter } = props;

  function handleForm(event) { event.preventDefault(); }

  const cat = ['900000000000', '100000000000', '50000000000'];
  return (
    <select name="category" id="category" className="categories" onSubmit={handleForm} value={filt} ref={category} onChange={() => filter(category.current.value)}>
      <option value="All" key="All">All</option>
      {cat.map((catt) => <option value={catt} key={catt}>{catt}</option>)}
    </select>
  );
}

Filter.propTypes = { filt: PropTypes.instanceOf(Function).isRequired };
Filter.propTypes = { filter: PropTypes.instanceOf(Function).isRequired };

export default Filter;
