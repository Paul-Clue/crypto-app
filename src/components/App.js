import React, { useEffect } from 'react';
import '../assets/stylesheets/App.css';
import { useSelector, useDispatch } from 'react-redux';
import { addCoin } from '../actions';
import Coin from './Coin';
import getData from '../util/apiFetch';

function App() {
  const coinList = useSelector((state) => state.addCoin);
  const filtered = useSelector((state) => state.filter);
  const filteredCoins = coinList.filter((coin) => coin.marketCap <= parseInt(filtered, 10) || filtered === 'All');
  const dispatch = useDispatch();

  const url = 'https://api.coinstats.app/public/v1/coins?skip=0&limit=10';

  useEffect(() => {
    getData(url)
      .then((response) => response.json())
      .then((result) => {
        dispatch(addCoin(result.coins));
      });
  }, []);

  return (
    <div className="App">
      <ul className="UL">
        {filteredCoins.map((coin) => <Coin key={coin.name} coin={coin} />)}
      </ul>
    </div>
  );
}

export default App;
