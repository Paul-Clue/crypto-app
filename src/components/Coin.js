import React from 'react';
import PropTypes from 'prop-types';
import '../assets/stylesheets/Coin.css';
import { useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import { detailFilter } from '../actions';

function Coin(props) {
  const dispatch = useDispatch();
  const { coin } = props;
  const { name } = coin.name;

  function sendInfo(load) {
    dispatch(detailFilter(load));
  }

  return (
    <li className="item">
      <Link className="Link item" coin={name} onClick={() => sendInfo(coin)} to={`/Details/${coin.name}`}>
        <span>
          <h1>
            <img src={coin.icon} alt="coin" width="45px" />
            {coin.name}
          </h1>
        </span>
        <h1>
          Market Cap:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
          {coin.marketCap}
        </h1>
      </Link>
    </li>
  );
}

Coin.propTypes = { coin: PropTypes.instanceOf(Object).isRequired };

export default Coin;
