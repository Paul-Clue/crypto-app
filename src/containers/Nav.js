import React from 'react';
import { Link } from 'react-router-dom';
import { useSelector } from 'react-redux';
import Filter from './Filter';
import '../assets/stylesheets/Nav.css';

function Nav() {
  const filtered = useSelector((state) => state.filter);

  return (
    <div>
      <nav className="navbar navbar-light bg-info">
        <div className="container-fluid">
          <span className="navbar-brand mb-0 h1">
            <h1 className="NavHead">
              <span className="dollar">
                $$
              </span>
              CRYPTO
              <span className="dollar">
                $$
              </span>
            </h1>
          </span>
          <div>
            <Filter filt={filtered} />
            &nbsp;
            <span className="filtTag">Filter By Market Cap</span>
            &nbsp;
            <span className="filtTag2"> *Caps less than value chosen will appear</span>
            <Link className="Link home" to="/">
              Home
            </Link>
          </div>
        </div>
      </nav>
    </div>
  );
}

export default Nav;
