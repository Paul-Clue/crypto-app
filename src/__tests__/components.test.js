import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { configureStore, createStore, combineReducers } from '@reduxjs/toolkit';
import CoinsReducer from '../reducers/coin';
import FilterReducer from '../reducers/filter';
import DetailsReducer from '../Reducers/detail';
import rootReducer from '../Reducers/index';
import App from '../components/App';
import Details from '../components/Details';
import Coin from '../components/Coin';
import Error from '../components/Error';
import Filter from '../containers/Filter';
import Nav from '../containers/Nav';

const coins = [{
  name: 'bitcoin',
}];

const filters = 'All';

const details = [{
  name: 'testing',
}];

const mockStore = configureStore({
  reducer: { rootReducer }, addCoin: coins, filter: filters, detail: details,
});

describe('Nav Component', () => {
  test('It should render', () => {
    const nav = renderer.create(
      <Provider store={mockStore}>
        <BrowserRouter>
          <Nav />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(nav).toMatchSnapshot();
  });
});

class ErrorBoundary extends React.Component {
  constructor(props) {
    super(props);
    this.state = { hasError: false };
  }

  static getDerivedStateFromError() {
    return { hasError: true };
  }

  componentDidCatch(error, errorInfo) {
    console.log(error, errorInfo);
  }

  render() {
    if (this.state.hasError) {//eslint-disable-line
      return <h1>Something went wrong.</h1>;
    }

    return this.props.children;//eslint-disable-line
  }
}

describe('Filter Component', () => {
  test('It should render', () => {
    const filt = renderer.create(
      <BrowserRouter>
        <ErrorBoundary>
          <Filter />
        </ErrorBoundary>
      </BrowserRouter>,
    ).toJSON();
    expect(filt).toMatchSnapshot();
  });
});

describe('Coin Component', () => {
  test('It should render', () => {
    const coin = renderer.create(
      <BrowserRouter>
        <ErrorBoundary>
          <Coin />
        </ErrorBoundary>
      </BrowserRouter>,
    ).toJSON();
    expect(coin).toMatchSnapshot();
  });
});

describe('Error Component', () => {
  test('It should render', () => {
    const err = renderer.create(
      <BrowserRouter>
        <Error />
      </BrowserRouter>,
    ).toJSON();
    expect(err).toMatchSnapshot();
  });
});

export function createTestStore() {//eslint-disable-line
  const store = createStore(
    combineReducers({
      addCoin: CoinsReducer,
      filter: FilterReducer,
      detail: DetailsReducer,
    }),
  );
  return store;
}

let store;
describe('App Component', () => {
  beforeEach(() => {
    store = createTestStore();
  });
  test('It should render', () => {
    const home = renderer.create(
      <Provider store={store}>
        <App />
      </Provider>,
    ).toJSON();
    expect(home).toMatchSnapshot();
  });
});

describe('Details Component', () => {
  beforeEach(() => {
    store = createTestStore();
  });
  test('It should render', () => {
    const home = renderer.create(
      <Provider store={store}>
        <Details />
      </Provider>,
    ).toJSON();
    expect(home).toMatchSnapshot();
  });
});
