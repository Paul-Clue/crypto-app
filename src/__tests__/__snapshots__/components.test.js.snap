// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component It should render 1`] = `
<div
  className="App"
>
  <ul
    className="UL"
  />
</div>
`;

exports[`Coin Component It should render 1`] = `
<h1>
  Something went wrong.
</h1>
`;

exports[`Details Component It should render 1`] = `
<div
  className="detail"
>
  <div
    className="TitleName"
  >
    <img
      alt="coin logo"
    />
    <h1
      className="TitleName1"
    />
  </div>
  <h1
    className="detail2"
  >
    <span
      className="bullet"
    >
      <span
        className="bold"
        id="one"
      >
        Symbol:
      </span>
         
    </span>
    <span
      className="bullet"
    >
      <span
        className="bold"
      >
        Rank:
      </span>
         
    </span>
    <span
      className="bullet"
    >
      <span
        className="bold"
      >
        Price:
      </span>
         
    </span>
    <span
      className="bullet"
    >
      <span
        className="bold"
      >
        Market Cap:
      </span>
         
    </span>
    <span
      className="bullet"
    >
      <span
        className="bold"
      >
        Total Supply:
      </span>
         
    </span>
    <span
      className="bullet"
    >
      <span
        className="bold"
      >
        Price Change 1h:
      </span>
         
    </span>
    <span
      className="bullet"
    >
      <span
        className="bold"
      >
        Price Change 1d:
      </span>
         
    </span>
    <span
      className="bullet"
    >
      <span
        className="bold"
      >
        Price Change 1w:
      </span>
         
    </span>
  </h1>
</div>
`;

exports[`Error Component It should render 1`] = `
<h1>
  Oops! Page not found!
</h1>
`;

exports[`Filter Component It should render 1`] = `
<h1>
  Something went wrong.
</h1>
`;

exports[`Nav Component It should render 1`] = `
<div>
  <nav
    className="navbar navbar-light bg-info"
  >
    <div
      className="container-fluid"
    >
      <span
        className="navbar-brand mb-0 h1"
      >
        <h1
          className="NavHead"
        >
          <span
            className="dollar"
          >
            $$
          </span>
          CRYPTO
          <span
            className="dollar"
          >
            $$
          </span>
        </h1>
      </span>
      <div>
        <select
          className="categories"
          id="category"
          name="category"
          onChange={[Function]}
          onSubmit={[Function]}
        >
          <option
            value="All"
          >
            All
          </option>
          <option
            value="900000000000"
          >
            900000000000
          </option>
          <option
            value="100000000000"
          >
            100000000000
          </option>
          <option
            value="50000000000"
          >
            50000000000
          </option>
        </select>
         
        <span
          className="filtTag"
        >
          Filter By Market Cap
        </span>
         
        <span
          className="filtTag2"
        >
           *Caps less than value chosen will appear
        </span>
        <a
          className="Link home"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </div>
    </div>
  </nav>
</div>
`;
